File: Programmers/Lv1/달리기 경주.py
def solution(players, callings):
    player_dict = {player: idx for idx, player in enumerate(players)}
    # enumerate() 사용 알기

    # print(player_dict)
    for name in callings:
        idx = player_dict[name]
        players[idx], players[idx - 1] = players[idx - 1], players[idx]

        player_dict[players[idx]] = idx
        player_dict[players[idx - 1]] = idx - 1

    return players

File: Programmers/Lv1/달리기 경주.js
function solution(players, callings) {
    var rank = {};
    cnt = 0
    for(name of players){
        rank[name] = cnt;
        cnt += 1;
    }
    
    for(call of callings){
        let idx = rank[call];
        let front = players[idx - 1];
        // 한 칸 앞에
        
        players[idx - 1] = call;
        players[idx] = front
        
        rank[call] = idx - 1;
        rank[front] = idx;
    }
    return players;
}